Add default database settting, and env var overrides

tankyleo opened the pull request:
Based on #72

tnull commented on rust/server/src/util/config.rs:
```diff
@@ -1,58 +1,142 @@
-
-		format!("postgresql://{}:{}@{}:{}", username, password, self.host, self.port)
-	}
+pub(crate) struct Configuration {
```
So now we have `Config` *and* `Configuration`? That seems rather confusing. Can they be merged, or at least one renamed appropriately to better distinguish their intended use?

tnull commented on rust/server/src/util/config.rs:
```diff
@@ -1,58 +1,142 @@
+		toml::from_str(&config_file)
+			.map_err(|e| format!("Failed to parse configuration file: {}", e))?;
+
+	macro_rules! read_env {
```
Do we really need macros for this? Maybe we can inline these or make them methods?

tnull commented on rust/server/src/util/config.rs:
```diff
@@ -11,8 +11,10 @@ const PSQL_VSS_DB_VAR: &str = "VSS_PSQL_VSS_DB";
+// environment variable.
 #[derive(Deserialize, Default)]
-struct Config {
+struct TomlConfigTable {
```
nit: When you squash, can we drop the `Table` here?

tnull reviewed (commented):
> Thanks tnull addressed your comments let me know if I can rebase

Please go ahead and squash & rebase. While you're at it, you might address the one nit below.

tankyleo commented:
Rebased and squashed with the new `TomlConfig` change, and the change noted in the last commit message:
```
If a file is passed as an argument, and it does not exist, we now abort startup.
```
